packages/serverpod/lib/src/server/tasks/task_manager.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'task.dart';
+
+/// Manages tasks that need to be executed.
+///
+/// The `TaskManager` allows registering tasks that can be executed
+/// concurrently. Each task is identified by a unique ID and contains
```

Comment:
The implementation and documentation here is super clear, so I guess it just changed from the PR description:

> Listeners are executed sequentially and awaited properly


If I recall correctly, that was initially planned (also with Task dependencies, right?).

So we might just need to update the description then.

Personally I would've probably started out with sequential execution, as I would imagine the tasks being rather short and at least then the developer has a "guarantee" that it would work the next time as well.

On the other hand, if they do have dependent shutdown operations, they can combine them in a bigger task.